### PR TITLE
[Enhancement] Remove final merge in load of PrimaryKey table

### LINKED ISSUE #10953: Remove final merge of PrimaryKey table to speed up load
## Enhancement
PrimaryKey table will write data into temp files first, then read and merge temp files in data load right now. So PrimaryKey table need to read and write all data twice in data load which cause the load time is twice as long as another types of tables. 
Actually, final merge of PrimaryKey table is not necessary, we can remove final merge to speed up data load.

### PR DESCRIPTION
**Pull request by sevev**
## What type of PR is this：
- [ ] bugfix
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes https://github.com/StarRocks/starrocks/issues/10953

## Problem Summary(Required) ：

PrimaryKey table will write data into temp files first, then read and merge temp files in data load. So PrimaryKey table needs to read and write all data twice in data load, which causes the load time to be twice as long as other types of tables.

The final merge of the PrimaryKey table is unnecessary; we can remove the final merge to speed up the data load.

In my test environment, the load cost time is shown as below:

| branch | load_type | num_rows | format | file size | key type | cost time | final merge time | 
| ------- | ---------- | ---------- | ------- | -------  | -------- | ---------  | ---------------- |
|   main   | stream load | 20000000 | CSV | 7.6G | PrimaryKey | 205668ms |      102151ms  |
| remove_final_merge| stream load | 20000000 | CSV | 7.6G | PrimaryKey| 97577ms | |

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by sevev** [comment]
run starrocks_admit_test

File: be/src/storage/rowset/beta_rowset_writer.cpp [hunk 1]
```diff
@@ -379,11 +381,14 @@ StatusOr<RowsetSharedPtr> HorizontalBetaRowsetWriter::build() {
...
-    if (_num_segment == 1) {
-        RETURN_IF_ERROR_WITH_WARN(
-                _fs->rename_file(Rowset::segment_temp_file_path(_context.rowset_path_prefix, _context.rowset_id, 0),
-                                 Rowset::segment_file_path(_context.rowset_path_prefix, _context.rowset_id, 0)),
-                "Fail to rename file");
+    if (_num_segment == 1 || !_context.schema_change_sorting) {
+        for (size_t seg_id = 0; seg_id < _num_segment; seg_id++) {
+            RETURN_IF_ERROR_WITH_WARN(
+                    _fs->rename_file(
+                            Rowset::segment_temp_file_path(_context.rowset_path_prefix, _context.rowset_id, seg_id),
```
Comment (trueeyu) [inline be/src/storage/rowset/beta_rowset_writer.cpp:388]: No need to rename file

**Review by trueeyu — DISMISSED** [review]

File: be/src/storage/rowset/beta_rowset_writer.cpp (hunk 1 above)
Comment (sevev) [reply be/src/storage/rowset/beta_rowset_writer.cpp]: > No need to rename file

done, changed in https://github.com/StarRocks/starrocks/pull/10903/commits/04a55a7ee3f427c68ea05b8f7cfaa7c092b3dd2d

**Comment by sevev** [comment]
run starrocks_admit_test

**Comment by sevev** [comment]
run starrocks_admit_test

**Comment by sevev** [comment]
run starrocks_admit_test

**Comment by sevev** [comment]
run starrocks_admit_test

**Comment by sevev** [comment]
run all

**Comment by sevev** [comment]
run starrocks_admit_test

**Comment by sevev** [comment]
run starrocks_admit_test

**Comment by sevev** [comment]
run starrocks_admit_test

**Review by trueeyu — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by chaoyli** [comment]
@mergify backport branch-2.4

**Pull request merged**